diff --git a/subsys/nrf_security/cmake/psa_crypto_want_config.cmake b/subsys/nrf_security/cmake/psa_crypto_want_config.cmake
index 5ca0955fda..dd4ab77485 100644
--- a/subsys/nrf_security/cmake/psa_crypto_want_config.cmake
+++ b/subsys/nrf_security/cmake/psa_crypto_want_config.cmake
@@ -12,6 +12,7 @@ kconfig_check_and_set_base_to_one(MBEDTLS_ASN1_WRITE_C)
 kconfig_check_and_set_base_to_one(MBEDTLS_BASE64_C)
 kconfig_check_and_set_base_to_one(MBEDTLS_MD_C)
 kconfig_check_and_set_base_to_one(MBEDTLS_MEMORY_BUFFER_ALLOC_C)
+kconfig_check_and_set_base_to_one(MBEDTLS_MEMORY_DEBUG)
 kconfig_check_and_set_base_int(MBEDTLS_MPI_MAX_SIZE)
 kconfig_check_and_set_base_int(MBEDTLS_MPI_WINDOW_SIZE)
 kconfig_check_and_set_base_to_one(MBEDTLS_PEM_PARSE_C)
diff --git a/subsys/nrf_security/configs/psa_crypto_want_config.h.template b/subsys/nrf_security/configs/psa_crypto_want_config.h.template
index 5273cf1156..ffb72c7a29 100644
--- a/subsys/nrf_security/configs/psa_crypto_want_config.h.template
+++ b/subsys/nrf_security/configs/psa_crypto_want_config.h.template
@@ -16,6 +16,7 @@
 #cmakedefine MBEDTLS_BASE64_C
 #cmakedefine MBEDTLS_MD_C
 #cmakedefine MBEDTLS_MEMORY_BUFFER_ALLOC_C
+#cmakedefine MBEDTLS_MEMORY_DEBUG
 #cmakedefine MBEDTLS_MPI_MAX_SIZE          @MBEDTLS_MPI_MAX_SIZE@
 #cmakedefine MBEDTLS_MPI_WINDOW_SIZE       @MBEDTLS_MPI_WINDOW_SIZE@
 #cmakedefine MBEDTLS_PEM_PARSE_C
